sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestFramework/IssueLocationCollector.cs

```diff
@@ -208,29 +291,50 @@ private IEnumerable<string> GetIssueIds(Match match)
 
         private int GetOffset(Match match)
         {
-            var sign = match.Groups["sign"];
             var offset = match.Groups["offset"];
-            if (!sign.Success && !offset.Success)
+            return offset.Success ? int.Parse(offset.Value) : 0;
+        }
+
+        private static void EnsureNoDuplicatedPrimaryIds(IList<IIssueLocation> mergedLocations)
+        {
+            var duplicateLocationsIds = mergedLocations
+                            .Where(location => location.IsPrimary)
```

Comment:
This should be little cheaper. Two where's looks suspicious when the second one does not touch the grouped values.
```suggestion
                            .Where(x => x.IsPrimary && x.Key != null)
```